File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Matematicas/MCM_Array.java
import java.io.PrintWriter;
import java.util.*;

//Mínimo Común Múltiplo de Array
public class Main {
	
	static long LCMarray(long [] arr) {
		long ans = 1;
		int divisor = 2;
		while(true) {
			int cont = 0;
			boolean divisible = false;
			for (int i = 0; i < arr.length; i++) {
				if (arr[0] == 0){
					return 0;
				}else if (arr[i] < 0) {
					arr[i] = Math.abs(arr[i]);	
				}
				if (arr[i] == 1) {
					cont++;
				}
				if (arr[i]%divisor == 0) {
					divisible = true;
					arr[i] = arr[i] / divisor;
				}
			}
			if(divisible == true) {
				ans = ans * divisor;
			}else {
				divisor++;
			}
			if(cont==arr.length) {
				return ans;
			}
		}
	}
	
	public static void main(String[] args) {
		long arr[] = {15, 12, 18};
		System.out.println(LCMarray(arr));
		
	}
}

File: Codigos estudiantes por lenguaje/C++/Daniel Cangrejo/Matematicas/MCM_Array.cpp
#include<bits/stdc++.h>
#include<cstdlib>
using namespace std;
typedef long long int ll;
ll LCMarray(ll arr[],int n){
    ll ans=1;
    ll divisor=2;
    while(true){
        int counter=0;
        bool divisible=false;
        for(int i=0;i<n;i++){
            if(arr[0]==0){
                return 0;
            }else if(arr[i]<0){
                arr[i]=abs(arr[i]);
            }
            if(arr[i]==1){
                counter++;
            }
            if(arr[i]%divisor==0){
                divisible=true;
                arr[i]/=divisor;
            }
        }
        if(divisible){
            ans*=divisor;
        }else{
            divisor++;
        }
        if(counter==n){
            return ans;
        }
    }
}
int main() {
    ll arr[]={2,7,3,9,4};
    ll n=sizeof(arr)/sizeof(arr[0]);
    cout<<LCMarray(arr,n)<<endl;
    return 0;
}
